qemu/tests/boot_with_different_vectors.py

```diff
@@ -45,41 +49,168 @@ def boot_guest_with_vectors(vectors):
-            d_id = device.split()[0]
-            msi_check_cmd = params["msi_check_cmd"] % d_id
-            output = session.cmd_output(msi_check_cmd)
-            if vectors == 0 and output:
-                test.fail("Guest do not support msi when vectors = 0.")
-            if output:
-                if vectors == 1:
-                    if "MSI-X: Enable-" in output:
-                        logging.info("MSI-X is disabled")
-                    else:
-                        msg = "Command %s get wrong output." % msi_check_cmd
-                        msg += " when vectors = 1"
-                        test.fail(msg)
-                else:
-                    if "MSI-X: Enable+" in output:
-                        logging.info("MSI-X is enabled")
+        if params["os_type"] == "linux":
+            devices = session.cmd_output("lspci | grep Eth").strip()
+            error_context.context("Check if vnic inside guest support msi.",
+                                  logging.info)
+            for device in devices.split("\n"):
+                if not device:
+                    continue
+                d_id = device.split()[0]
+                msi_check_cmd = params["msi_check_cmd"] % d_id
+                output = session.cmd_output(msi_check_cmd)
+                if vectors == 0 and output:
+                    test.fail("Guest do not support msi when vectors = 0.")
+                if output:
+                    if vectors == 1:
+                        if "MSI-X: Enable-" in output:
+                            logging.info("MSI-X is disabled")
+                        else:
+                            msg = "Command %s get wrong output." % msi_check_cmd
+                            msg += " when vectors = 1"
+                            test.fail(msg)
                     else:
-                        msg = "Command %s get wrong output." % msi_check_cmd
-                        msg += " when vectors = %d" % vectors
-                        test.fail(msg)
+                        if "MSI-X: Enable+" in output:
+                            logging.info("MSI-X is enabled")
+                        else:
+                            msg = "Command %s get wrong output." % msi_check_cmd
+                            msg += " when vectors = %d" % vectors
+                            test.fail(msg)
+        else:
+            # use traceview
+            prepare_traceview_windows(session, 360)
+            msis, queues = get_MSIs_and_queues_windows()
+            if msis is None or queues is None:
+                test.fail("Can't get msi status from guest.")
+            if vectors == 0 and msis != 0 and queues != 1:
+                test.fail("Msis should be 0, queues should be 1  when vectors = 1,"
+                          " but guest msis = %s, queues = %s" % (msis, queues))
+            elif vectors != msis:
+                test.fail("Msis should equal to vectors(%s), but guest is %s" % (vectors, msis))
```

Comment:
Could you add else conditon

Reply:
But the "else " is only pass this case. So I think it no need to add this condition